src/openloco/windows/IndustryWindow.cpp

```diff
@@ -0,0 +1,414 @@
+
+                flags = self->viewports[0]->flags;
+                self->viewports[0]->width = 0;
+                self->viewports[0] = nullptr;
+                viewportmgr::collectGarbage();
+            }
+            else
+            {
+                if ((config::get().flags & config::flags::gridlines_on_landscape) != 0)
+                    flags |= viewport_flags::gridlines_on_landscape;
+            }
+
+            self->saved_view = view;
+
+            // 0x00499B39 start
+            if (self->viewports[0] == nullptr)
+            {
+                auto widget = &self->widgets[widx::viewport];
+                auto tile = openloco::map::map_pos3({ industry->x, industry->y, tileZ });
+                auto origin = gfx::point_t(widget->left + self->x + 1, widget->top + self->y + 1);
+                auto size = gfx::ui_size_t(widget->width() - 2, widget->height() - 2);
+                viewportmgr::create(self, 0, origin, size, self->saved_view.zoomLevel, tile);
+                self->invalidate();
+                self->flags |= window_flags::viewport_no_scrolling;
+            }
+            // 0x00499B39 end
+
+            if (self->viewports[0] != nullptr)
+            {
+                self->viewports[0]->flags = flags;
+                self->invalidate();
+            }
+        }
+
+        static void initEvents()
+        {
+            events.draw = draw;
+            events.on_mouse_up = on_mouse_up;
+            events.on_resize = on_resize;
+            events.on_update = common::update;
+            events.prepare_draw = prepare_draw;
+            events.text_input = common::text_input;
+            events.viewport_rotate = initViewport;
+        }
+    }
+
+    // 0x00456D2D
+    window* open(uint16_t industryId)
+    {
+        auto window = WindowManager::bringToFront(WindowType::industry, industryId);
+        if (window != nullptr)
+        {
+            if (input::is_tool_active(window->type, window->number))
+                input::cancel_tool();
+
+            window = WindowManager::bringToFront(WindowType::industry, industryId);
+        }
+
+        if (window == nullptr)
+        {
+            // 0x00456DBC start
+            const uint32_t newFlags = window_flags::flag_8 | window_flags::resizable;
+            window = WindowManager::createWindow(WindowType::industry, windowSize, newFlags, &industry::events);
+            window->number = industryId;
+            window->min_width = 192;
+            window->min_height = 137;
+            window->max_width = 600;
+            window->max_height = 440;
+
+            auto skin = objectmgr::get<interface_skin_object>();
+            if (skin != nullptr)
+            {
+                window->colours[0] = skin->colour_0B;
+                window->colours[1] = skin->colour_0C;
+            }
+            // 0x00456DBC end
+
+            window->saved_view.clear();
+        }
+
+        // TODO(avgeffen): only needs to be called once.
```

Comment:
This is your line, not mine. You can leave out the name, though, as we don't really follow the convention in this project. Just write:
```suggestion
        // TODO: only needs to be called once.
```

Reply:
OK, I hadn't realised it was your name and was wondering what it stood for! I think I did the same in Station Window PR.